### PR TITLE
logging: adds papertrail endpoint

### PR DESCRIPTION
**Pull request by mccurdyc**
## Relevant (Referenced) Link(s)

* [Fastly API Docs](https://docs.fastly.com/api/logging#api-section-logging_papertrail)
* [fastly/go-fastly](https://github.com/fastly/go-fastly/blob/88b210b97743b0bfece81dc3dbbd7e3254e22d97/fastly/papertrail.go)

## Validation

```
$ make test
```

```
git clone git@github.com:mccurdyc/cli.git /tmp/cli && \
( 
    cd /tmp/cli && \
    git checkout mccurdyc/logging-papertrail && \
    make test && \
    rm -rf /tmp/cli \
)
```

### CODE REVIEW

File: pkg/logging/papertrail/create.go [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+	"github.com/fastly/cli/pkg/common"
+	"github.com/fastly/cli/pkg/compute/manifest"
+	"github.com/fastly/cli/pkg/config"
+	"github.com/fastly/cli/pkg/errors"
+	"github.com/fastly/cli/pkg/text"
+	"github.com/fastly/go-fastly/fastly"
+)
+
+// CreateCommand calls the Fastly API to create  Papertrail logging endpoints.
```
Comment (phamann) [inline pkg/logging/papertrail/create.go:14]:
```suggestion
// CreateCommand calls the Fastly API to create Papertrail logging endpoints.
```

File: pkg/logging/papertrail/delete.go [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+	"github.com/fastly/cli/pkg/common"
+	"github.com/fastly/cli/pkg/compute/manifest"
+	"github.com/fastly/cli/pkg/config"
+	"github.com/fastly/cli/pkg/errors"
+	"github.com/fastly/cli/pkg/text"
+	"github.com/fastly/go-fastly/fastly"
+)
+
+// DeleteCommand calls the Fastly API to delete  Papertrail logging endpoints.
```
Comment (phamann) [inline pkg/logging/papertrail/delete.go:14]:
```suggestion
// DeleteCommand calls the Fastly API to delete  Papertrail logging endpoints.
```
```suggestion
// DeleteCommand calls the Fastly API to delete Papertrail logging endpoints.
```

File: pkg/logging/papertrail/describe.go [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+	"io"
+
+	"github.com/fastly/cli/pkg/common"
+	"github.com/fastly/cli/pkg/compute/manifest"
+	"github.com/fastly/cli/pkg/config"
+	"github.com/fastly/cli/pkg/errors"
+	"github.com/fastly/go-fastly/fastly"
+)
+
+// DescribeCommand calls the Fastly API to describe an  Papertrail logging endpoint.
```
Comment (phamann) [inline pkg/logging/papertrail/describe.go:14]:
```suggestion
// DescribeCommand calls the Fastly API to describe an Papertrail logging endpoint.
```

File: pkg/logging/papertrail/list.go [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+
+	"github.com/fastly/cli/pkg/common"
+	"github.com/fastly/cli/pkg/compute/manifest"
+	"github.com/fastly/cli/pkg/config"
+	"github.com/fastly/cli/pkg/errors"
+	"github.com/fastly/cli/pkg/text"
+	"github.com/fastly/go-fastly/fastly"
+)
+
+// ListCommand calls the Fastly API to list  Papertrail logging endpoints.
```
Comment (phamann) [inline pkg/logging/papertrail/list.go:15]:
```suggestion
// ListCommand calls the Fastly API to list Papertrail logging endpoints.
```

File: pkg/logging/papertrail/update.go [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+
+	"github.com/fastly/cli/pkg/common"
+	"github.com/fastly/cli/pkg/compute/manifest"
+	"github.com/fastly/cli/pkg/config"
+	"github.com/fastly/cli/pkg/errors"
+	"github.com/fastly/cli/pkg/text"
+	"github.com/fastly/go-fastly/fastly"
+)
+
+// UpdateCommand calls the Fastly API to update  Papertrail logging endpoints.
```
Comment (phamann) [inline pkg/logging/papertrail/update.go:14]:
```suggestion
// UpdateCommand calls the Fastly API to update Papertrail logging endpoints.
```

File: pkg/logging/papertrail/root.go [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+type RootCommand struct {
+	common.Base
+	// no flags
+}
+
+// NewRootCommand returns a new command registered in the parent.
+func NewRootCommand(parent common.Registerer, globals *config.Data) *RootCommand {
+	var c RootCommand
+	c.Globals = globals
+	c.CmdClause = parent.Command("papertrail", "Manipulate Fastly service version Papertrail logging endpoints.")
```
Comment (phamann) [inline pkg/logging/papertrail/root.go:21]:
```suggestion
	c.CmdClause = parent.Command("papertrail", "Manipulate Fastly service version Papertrail logging endpoints")
```

File: pkg/logging/papertrail/create.go [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+	manifest manifest.Data
+	Input    fastly.CreatePapertrailInput
+}
+
+// NewCreateCommand returns a usable command registered under the parent.
+func NewCreateCommand(parent common.Registerer, globals *config.Data) *CreateCommand {
+	var c CreateCommand
+	c.Globals = globals
+	c.manifest.File.Read(manifest.Filename)
+	c.CmdClause = parent.Command("create", "Create an  Papertrail logging endpoint on a Fastly service version").Alias("add")
```
Comment (phamann) [inline pkg/logging/papertrail/create.go:26]:
```suggestion
	c.CmdClause = parent.Command("create", "Create a Papertrail logging endpoint on a Fastly service version").Alias("add")
```

File: pkg/logging/papertrail/create.go [hunk 8]
```diff
@@ -0,0 +1,57 @@
...
+	var c CreateCommand
+	c.Globals = globals
+	c.manifest.File.Read(manifest.Filename)
+	c.CmdClause = parent.Command("create", "Create a Papertrail logging endpoint on a Fastly service version").Alias("add")
+
+	c.CmdClause.Flag("name", "The name of the Papertrail logging object. Used as a primary key for API access").Short('n').Required().StringVar(&c.Input.Name)
+	c.CmdClause.Flag("service-id", "Service ID").Short('s').StringVar(&c.manifest.Flag.ServiceID)
+	c.CmdClause.Flag("version", "Number of service version").Required().IntVar(&c.Input.Version)
+
+	c.CmdClause.Flag("address", "A hostname or IPv4 address").Required().StringVar(&c.Input.Address)
```
Comment (mccurdyc) [inline pkg/logging/papertrail/create.go:32]:
```suggestion
	c.CmdClause.Flag("address", "A hostname or IPv4 address").Required().StringVar(&c.Input.Address)
	
```

**Comment by mccurdyc** [comment]
Force-pushed rebasing master.

**Comment by mccurdyc** [comment]
Force-pushed rebasing into a single commit.

**Comment by mccurdyc** [comment]
Force-pushed a formatting fix and rebased into a single commit.

**Pull request merged**